### PR TITLE
Add option to specify IAM role name

### LINKED ISSUE #2261: Iamserviceaccount created role naming should match the role instead of RoleN
**Why do you want this feature?**
Along with the existing tagging feature it would be nice to have the role that is created have a distinguishable name related to the role.

**What feature/behavior/change do you want?**
Iamserviceaccount created role naming should match the role instead of RoleN

*** Example **
`$ eksctl create iamserviceaccount --cluster=breakfast --name glazeddougnuts --tags "Jira=WONKA-1234, Environment=dev" --namespace=glazeddougnuts --attach-policy-arn=arn:aws:iam::aws:policy/AmazonChimeSDK --profile breakfast --region us-east-1 --approve'

Create a role like
'breakfast-addon-iamserviceaccount-glazeddougnuts'

instead of 

'breakfast-addon-iamserviceaccoun-Role1-1NIFREKFVTZE2'

This can be resolved by using the RoleName: parameter in the cloudformation

### PR DESCRIPTION
**Pull request by Tolsto**
Currently, CloudFormation generates a role name that contains
a random suffix. However, in some use-cases a predetermined
role name is useful. For instance, when following a git-ops
workflow it might be necessary to include the serviceaccount
annotations for the IAM role ARNs in the manifest repo. With
random role names that can only be done after the role has
been created by eksctl. In addition, if the cluster should ever
get re-created all role ARNs would change which would then
require all manifests that reference those ARNs to be edited.

The naming of the role Role1 and the comment here https://github.com/weaveworks/eksctl/blob/fc1f61f27f0f888edec824e1d975803d5345cac0/pkg/cfn/builder/iam.go#L225 suggest that there might be more than one role in the stack. However, I cannot see when this would be the case. 

Closes #2261

### Checklist
- [x] Added tests that cover your change (if possible)
- [x] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] Manually tested
- [ ] Added labels for change area (e.g. `area/nodegroup`), target version (e.g. `version/0.12.0`) and kind (e.g. `kind/improvement`)
- [ ] Make sure the title of the PR is a good description that can go into the release notes

### CODE REVIEW

**Comment by aclevername** [comment]
@Tolsto thanks for the PR!

The use case sounds reasonable. One slight drawback of this is that you lose the convenience of easily being able to spot who created the role, for example if I go and look at my roles in AWS I can clearly see that the roles were created by eksctl for a given cluster.

What do you think about changing it so you can specify the suffix instead? You will still get a consistent role name output, but you would retain the above value. Thoughts?

**Comment by Tolsto** [comment]
I originally intended to implement it with a suffix but decided against it because it leaves the user with less flexibility. The proposed change gives users the option to choose any role name they like, including a name that reflects the cluster name.

**Comment by aclevername** [comment]
Fixed some merge conflicts. 

Tested with the following config:
```
---
apiVersion: eksctl.io/v1alpha5
kind: ClusterConfig

metadata:
  name: jk-managed
  region: us-west-2

iam:
  withOIDC: true
  serviceAccounts:
  - metadata:
      name: s3-reader-no-role
      # if no namespace is set, "default" will be used;
      # the namespace will be created if it doesn't exist already
      namespace: backend-apps
      labels: {aws-usage: "application"}
    attachPolicyARNs:
    - "arn:aws:iam::aws:policy/AmazonS3ReadOnlyAccess"
  - metadata:
      name: s3-reader-with-role
      # if no namespace is set, "default" will be used;
      # the namespace will be created if it doesn't exist already
      namespace: backend-apps
      labels: {aws-usage: "application"}
    attachPolicyARNs:
      - "arn:aws:iam::aws:policy/AmazonS3ReadOnlyAccess"
    roleName: s3-reader-with-role
```

This produced the following:
```
$k get serviceaccounts -n backend-apps  -o yaml | grep amazonaws
      eks.amazonaws.com/role-arn: arn:aws:iam::083751696308:role/eksctl-jk-managed-addon-iamserviceaccount-ba-Role1-13NO2GL1J1C66
            f:eks.amazonaws.com/role-arn: {}
      eks.amazonaws.com/role-arn: arn:aws:iam::083751696308:role/s3-reader-with-role
            f:eks.amazonaws.com/role-arn: {}
 ```
 
 Looking in cloudformation all looks good:
 

Screenshot 2020-12-17 at 15 17 29 

Screenshot 2020-12-17 at 15 17 50

File: pkg/ctl/create/iamserviceaccount.go [hunk 1]
```diff
@@ -45,6 +45,7 @@ func createIAMServiceAccountCmdWithRunFunc(cmd *cmdutils.Cmd, runFunc func(cmd *
 		fs.StringVar(&serviceAccount.Name, "name", "", "name of the iamserviceaccount to create")
 		fs.StringVar(&serviceAccount.Namespace, "namespace", "default", "namespace where to create the iamserviceaccount")
 		fs.StringSliceVar(&serviceAccount.AttachPolicyARNs, "attach-policy-arn", []string{}, "ARN of the policy where to create the iamserviceaccount")
+		fs.StringVar(&serviceAccount.RoleName, "rolename", "", "Set a custom name for the created role")
```
Comment (Callisto13) [inline pkg/ctl/create/iamserviceaccount.go:48]: Tiny nit: should this flag be `--role-name` to be in line with our other more-than-one word flags?

**Review by Callisto13 — APPROVED** [review]

**Comment by Legion2** [comment]
IAM Roles created using this feature can not be attached to existing service accounts using the annotation. When trying to do so I get the error `WebIdentityErr: failed to retrieve credentials
caused by: AccessDenied: Not authorized to perform sts:AssumeRoleWithWebIdentity`. I think it's because the IAM Role have a Contidtion attached that must match the defined service account namespace and name. So when the role is attached to another service account it will not work.

**Comment by aclevername** [comment]
> IAM Roles created using this feature can not be attached to existing service accounts using the annotation. When trying to do so I get the error `WebIdentityErr: failed to retrieve credentials caused by: AccessDenied: Not authorized to perform sts:AssumeRoleWithWebIdentity`. I think it's because the IAM Role have a Contidtion attached that must match the defined service account namespace and name. So when the role is attached to another service account it will not work.

Hey @Legion2 👋 , When we create the IAMServiceAccount we add a sub condition to the Role to ensure that it can only be used by the specified service account https://github.com/weaveworks/eksctl/blob/5145b139fad61106beee3201ae62b6fb2be96abe/pkg/iam/oidc/api.go#L148-L156

`IAM Roles created using this feature can not be attached to existing service accounts using the annotation` I've tried it out and it appears to work for me:
```
k get serviceaccount -n default test-sa -o yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  creationTimestamp: "2021-01-07T11:47:38Z"
  name: test-sa
  namespace: default
  resourceVersion: "200311"
  selfLink: /api/v1/namespaces/default/serviceaccounts/test-sa
  uid: 031e2e7f-abea-40af-b7e9-32798a735393
secrets:
- name: test-sa-token-z9wqq

cat cluster.yaml
---
apiVersion: eksctl.io/v1alpha5
kind: ClusterConfig

metadata:
  name: jk-console
  region: us-west-2

iam:
  withOIDC: true
  serviceAccounts:
    - metadata:
        name: test-sa
        # if no namespace is set, "default" will be used;
        # the namespace will be created if it doesn't exist already
        namespace: default
        labels: {aws-usage: "application"}
      attachPolicyARNs:
        - "arn:aws:iam::aws:policy/AmazonS3ReadOnlyAccess"
      roleName: test-sa-role
      
 ./eksctl create iamserviceaccount -f cluster.yaml --override-existing-serviceaccounts --approve
[ℹ]  eksctl version 0.37.0-dev+5c484872.2021-01-07T11:46:15Z
[ℹ]  using region us-west-2
[ℹ]  1 iamserviceaccount (default/test-sa) was included (based on the include/exclude rules)
[!]  metadata of serviceaccounts that exist in Kubernetes will be updated, as --override-existing-serviceaccounts was set
[ℹ]  1 task: { 2 sequential sub-tasks: { create IAM role for serviceaccount "default/test-sa", create serviceaccount "default/test-sa" } }
[ℹ]  building iamserviceaccount stack "eksctl-jk-console-addon-iamserviceaccount-default-test-sa"
[ℹ]  deploying stack "eksctl-jk-console-addon-iamserviceaccount-default-test-sa"
[ℹ]  waiting for CloudFormation stack "eksctl-jk-console-addon-iamserviceaccount-default-test-sa"
[ℹ]  waiting for CloudFormation stack "eksctl-jk-console-addon-iamserviceaccount-default-test-sa"
[ℹ]  waiting for CloudFormation stack "eksctl-jk-console-addon-iamserviceaccount-default-test-sa"
[ℹ]  serviceaccount "default/test-sa" already exists
[ℹ]  updated serviceaccount "default/test-sa"

k get serviceaccount -n default test-sa -o yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  annotations:
    eks.amazonaws.com/role-arn: arn:aws:iam::<redacted>:role/test-sa-role
  creationTimestamp: "2021-01-07T11:47:38Z"
  labels:
    aws-usage: application
  name: test-sa
  namespace: default
  resourceVersion: "200946"
  selfLink: /api/v1/namespaces/default/serviceaccounts/test-sa
  uid: 031e2e7f-abea-40af-b7e9-32798a735393
secrets:
- name: test-sa-token-z9wqq
```

Could you provide more details about how you are encountering this problem in a new issue please 😄

**Comment by Legion2** [comment]
@aclevername see #3042

**Pull request merged**